src/entities/repos/institutions_repo.py

```diff
@@ -68,21 +71,24 @@ async def get_federal_regulators(session: AsyncSession) -> List[FederalRegulator
     return await query_type(session, FederalRegulatorDao)
 
 
-async def upsert_institution(session: AsyncSession, fi: FinancialInstitutionDto) -> FinancialInstitutionDao:
+async def upsert_institution(
+    session: AsyncSession, fi: FinancialInstitutionDto, user: AuthenticatedUser
+) -> FinancialInstitutionDao:
     async with session.begin():
         fi_data = fi.__dict__.copy()
         fi_data.pop("_sa_instance_state", None)
+        fi_data.pop("version", None)
 
         if "sbl_institution_types" in fi_data:
             types_association = [
-                SblTypeMappingDao(type_id=t)
+                SblTypeMappingDao(type_id=t, lei=fi.lei, modified_by=user.id)
                 if isinstance(t, str)
-                else SblTypeMappingDao(type_id=t.id, details=t.details)
+                else SblTypeMappingDao(type_id=t.id, details=t.details, lei=fi.lei, modified_by=user.id)
                 for t in fi.sbl_institution_types
             ]
             fi_data["sbl_institution_types"] = types_association
 
-        db_fi = await session.merge(FinancialInstitutionDao(**fi_data))
+        db_fi = await session.merge(FinancialInstitutionDao(**fi_data, modified_by=user.id, event_time=datetime.now()))
```

Comment:
It's six of one, but you can also specify on the DAO event_time field onupdate=func.now().